Role generation

ThomasKopsch opened the pull request:
I introduced different roles:
Now there is a new abstract class User from which the classes Customer and banker inherit. This will allow us to implement different functionality for different roles.

I suggest to implement a command SHOWACCOUNTS <CustomerID> which allows bankers to see the account of any customer.

ThomasKopsch committed:
Role generation

I introduced different roles:
Now there is a new abstract class User from which the classes Customer and banker inherit. This will allow us to implement different functionality for different roles.

I suggest to implement a command SHOWACCOUNTS <CustomerID> which allows bankers to see the account of any customer.

vc409 reviewed (commented):
Hi @ThomasKopsch ,

I have checked the code and I have the following comments:
- Please use CamelCase Java convention, instead of banker use Banker, etc.
- I think the method getUserType in the abstract class called User should be abstract so that you force the "child" to implement it.
- The class User defines a method setNewPassword where it compares the strings based on their JVM reference, if you wanted to compare the string content please use the **.equals** method of class String, and place the boolean expression in a single return, instead of an if-else, something like **return s1.equals(s2);**
- The class User can omit the constructor and the fields could be protected.

kelinj commented on src/newbank/server/User.java:
```diff
@@ -0,0 +1,42 @@
+	}
+	
+	public String getUserType() {
+		return "generic";
```
We can have this as abstract method. 
	`abstract public String getUserType() ;`
and not let a "generic" user type exist. The type will be implemented by the child classes

ThomasKopsch replied:
Ok, I agree.

kelinj commented on src/newbank/server/banker.java:
```diff
@@ -0,0 +1,21 @@
+
+import java.util.ArrayList;
+
+public class banker extends User {
```
Preferred to have Title case for Classes

ThomasKopsch replied:
Agree.

kelinj commented on src/newbank/server/userDB.java:
```diff
@@ -5,49 +5,49 @@
 import java.util.HashMap;
 
-public class customerDB {
+public class userDB {
```
Nice work generalizing this. 
 Preferred to use Title case for class

ThomasKopsch replied:
Agree.

kelinj commented on src/newbank/server/NewBank.java:
```diff
@@ -4,14 +4,14 @@
-	public synchronized Customer checkLogInDetails(String customerID, String password) {
-		Customer customer = customers.getCustomer(customerID);
+	public synchronized User checkLogInDetails(String userID, String password) {
+		User customer = users.getUser(userID);
```
`User user= users.getUser(userID);` 
Since generic

ThomasKopsch replied:
Agree.

kelinj commented on src/newbank/server/NewBank.java:
```diff
@@ -22,15 +22,30 @@ public synchronized Customer checkLogInDetails(String customerID, String passwor
+				}
+			//Execute SHOWACCOUNTS if the user is a customer			
+			} else if (request.startsWith("SHOWACCOUNTS")) {
+				if (user.getUserType().equals("banker")) {
```
Yeah true.  I can see a lot more like this which can be put in backlog specific to Banker 
Depositing money also could be something which can be Banker specific

ThomasKopsch committed:
Implemented suggestions from review

Implemented review suggestions:
- Use Camel Case for the class banker and userDB
- make method getUserType() in User.java abstrat
- Use .equals() function to compare two strings in User.java
